mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -429,65 +432,50 @@ static void correctParameters() {
   std::string filterLayoutValue = filterLayout.getValue();
   std::string inputLayoutValue = inputLayout.getValue();
   std::string outputLayoutValue = outputLayout.getValue();
-  if (filterLayoutValue.size() == 4) { // yxcgk not implement yet
-    if (filterLayoutValue == "kcyx")
-      filterLayout.setValue("gkcyx");
-    else if (filterLayoutValue == "kyxc")
-      filterLayout.setValue("gkyxc");
-    else
-      filterLayout.setValue("g" + filterLayoutValue);
-  }
-  if (outputLayoutValue.size() == 4) {
-    if (outputLayoutValue == "nkhw")
-      outputLayout.setValue("ngkhw");
-    else if (outputLayoutValue == "nhwk")
-      outputLayout.setValue("nhwgk");
-    else
-      outputLayout.setValue("g" + outputLayoutValue);
-  }
-
-  if (inputLayoutValue.size() == 4) {
-    if (inputLayoutValue == "nchw")
-      inputLayout.setValue("ngchw");
-    else if (inputLayoutValue == "nhwc")
-      inputLayout.setValue("nhwgc");
-    else
-      inputLayout.setValue("g" + inputLayoutValue);
-  }
-
-  // we can use paddingHeight or paddingHeightLeft + paddingHeightRight
-  // if use paddingHeight , paddingHeightLeft and paddingHeightRight =
-  // paddingHeight if use paddingHeightLeft + paddingHeightRight , please
-  // assigne value
-  if (paddingHeight.getValue() > 0) {
-    if (paddingHeightLeft.getValue() == 0 &&
-        paddingHeightRight.getValue() == 0) {
-      paddingHeightLeft.setValue(paddingHeight.getValue());
-      paddingHeightRight.setValue(paddingHeight.getValue());
-    } else {
-      if (paddingHeightLeft.getValue() != paddingHeight.getValue() ||
-          paddingHeightRight.getValue() != paddingHeight.getValue()) {
-        llvm::errs()
-            << "you can't use both padding_h and (padding_h_l,padding_h_r).\n";
+  // yxcgk not implement yet
```

Comment:
Cleanup and abstraction.  Checking for length-4 strings is redundant.  The padding checks are the same for both groups.